### ISSUE TITLE
fatal error: 'boost/mpi.hpp' file not found on MacOS

### ISSUE THREAD

**Issue by linonetwo** [microsoft/vcpkg#6640]
```vcpkg install boost``` won't install boost.MPI

And If I run `vcpkg install boost-mpi`, the result will be:

```
% vcpkg install boost-mpi
The following packages will be built and installed:
    boost-mpi[core]:x64-osx
  * msmpi[core]:x64-osx
Additional packages (*) will be modified to complete this operation.
Starting package 1/2: msmpi:x64-osx
Building package msmpi[core]:x64-osx...
-- Downloading https://download.microsoft.com/download/A/E/0/AE002626-9D9D-448D-8197-1EA510E297CE/msmpisdk.msi...
-- Downloading https://download.microsoft.com/download/A/E/0/AE002626-9D9D-448D-8197-1EA510E297CE/msmpisetup.exe...
CMake Error at ports/msmpi/portfile.cmake:57 (message):
    Could not find:
      /Microsoft MPI/Bin/mpiexec.exe
    Please install the MSMPI redistributable package before trying to install this port.
    The appropriate installer has been downloaded to:
      /Users/linonetwo/repo/vcpkg/downloads/msmpisetup-10.0.12498.exe

Call Stack (most recent call first):
  scripts/ports.cmake:71 (include)


Error: Building package msmpi:x64-osx failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: msmpi:x64-osx
  Vcpkg version: 2018.11.23-unknownhash

Additionally, attach any relevant sections from the log files above.
```

It try to run `msmpisetup.exe` in MacOS...

**Comment by wangli28** [comment]
#3242

**Comment by wangli28** [comment]
Hi @linonetwo, Thanks for posting this issue. You should manually install `msmpisetup-10.0.12498.exe` in this directory `/Users/linonetwo/repo/vcpkg/downloads` before installing `boost-mpi`.

    Please install the MSMPI redistributable package before trying to install this port.
    The appropriate installer has been downloaded to:
      /Users/linonetwo/repo/vcpkg/downloads/msmpisetup-10.0.12498.exe

**Comment by linonetwo** [comment]
But it will be hard to run `exe` file on MacOS.

But I still give it a try:

```
% ./msmpisetup-10.0.12498.exe 
zsh: permission denied: ./msmpisetup-10.0.12498.exe
% chmod +x ./msmpisetup-10.0.12498.exe    
% ./msmpisetup-10.0.12498.exe         
zsh: exec format error: ./msmpisetup-10.0.12498.exe
```

**Comment by yurybura** [comment]
@PhoebeHui Probably this issue was fixed in #7142

**Comment by PhoebeHui** [comment]
Yes, I installed it with latest source, it built successfully.
```
Starting package 3/3: boost-mpi:x64-osx
Building package boost-mpi[core]:x64-osx...
-- Downloading https://github.com/boostorg/mpi/archive/boost-1.76.0.tar.gz -> boostorg-mpi-boost-1.76.0.tar.gz...
-- Extracting source /Users/vcpkg/phoebe/vcpkg/downloads/boostorg-mpi-boost-1.76.0.tar.gz
-- Using source at /Users/vcpkg/phoebe/vcpkg/buildtrees/boost-mpi/src/ost-1.76.0-dcca087947.clean
-- Building release...
-- Configuring x64-osx-rel
CMake Warning at scripts/cmake/vcpkg_configure_cmake.cmake:433 (message):
  The following variables are not used in CMakeLists.txt:
      FEATURES
  Please recheck them and remove the unnecessary options from the
  `vcpkg_configure_cmake` call.

  If these options should still be passed for whatever reason, please use the
  `MAYBE_UNUSED_VARIABLES` argument.
Call Stack (most recent call first):
  installed/x64-osx/share/boost-build/boost-modular-build.cmake:88 (vcpkg_configure_cmake)
  installed/x64-osx/share/boost-build/boost-modular-build.cmake:111 (unix_build)
  ports/boost-mpi/portfile.cmake:15 (boost_modular_build)
  scripts/ports.cmake:140 (include)
-- Building x64-osx-rel
-- Building debug...
-- Configuring x64-osx-dbg
CMake Warning at scripts/cmake/vcpkg_configure_cmake.cmake:433 (message):
  The following variables are not used in CMakeLists.txt:
      FEATURES
  Please recheck them and remove the unnecessary options from the
  `vcpkg_configure_cmake` call.

  If these options should still be passed for whatever reason, please use the
  `MAYBE_UNUSED_VARIABLES` argument.
Call Stack (most recent call first):
  installed/x64-osx/share/boost-build/boost-modular-build.cmake:88 (vcpkg_configure_cmake)
  installed/x64-osx/share/boost-build/boost-modular-build.cmake:119 (unix_build)
  ports/boost-mpi/portfile.cmake:15 (boost_modular_build)
  scripts/ports.cmake:140 (include)
-- Building x64-osx-dbg
-- Packaging headers
-- Packaging headers done
-- Downloading https://raw.githubusercontent.com/boostorg/boost/boost-1.74.0/LICENSE_1_0.txt -> boost_LICENSE_1_0.txt...
-- Installing: /Users/vcpkg/phoebe/vcpkg/packages/boost-mpi_x64-osx/share/boost-mpi/usage
-- Installing: /Users/vcpkg/phoebe/vcpkg/packages/boost-mpi_x64-osx/share/boost-mpi/copyright
-- Performing post-build validation
-- Performing post-build validation done  
Stored binary cache: /Users/vcpkg/.cache/vcpkg/archives/1f/1fb407341f3d2e0c40b470c5427c99b3d258b7cfd87d8e1e8eeda8b51026cb9a.zip
Installing package boost-mpi[core]:x64-osx...
Elapsed time for package boost-mpi:x64-osx: 9.669 s

Total elapsed time: 25.7 min

The package boost is compatible with built-in CMake targets:
   
    find_package(Boost REQUIRED [COMPONENTS <libs>...])
    target_link_libraries(main PRIVATE Boost::boost Boost::<lib1> Boost::<lib2> ...)
```
